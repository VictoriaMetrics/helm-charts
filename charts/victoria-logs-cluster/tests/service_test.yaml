suite: service templates test
chart:
  version: 0.1.1
  appVersion: 0.1.0
templates:
  - service.yaml
tests:
  - it: service should match snapshot
    set:
      vmauth:
        enabled: true
    asserts:
      - matchSnapshot: {}
  - it: service should match snapshot with fullnameOverride and extraLabels
    set:
      vminsert:
        name: insert
        fullnameOverride: vminsert-node
        extraLabels:
          extraLabelName: insertExtraLabelValue
        service:
          labels:
            extraServiceLabelName: insertExtraServiceLabelValue
      vmauth:
        enabled: true
        name: auth
        fullnameOverride: vmauth-node
        extraLabels:
          extraLabelName: authExtraLabelValue
        service:
          labels:
            extraServiceLabelName: authExtraServiceLabelValue
      vmselect:
        name: select
        fullnameOverride: vmselect-node
        extraLabels:
          extraLabelName: selectExtraLabelValue
        service:
          labels:
            extraServiceLabelName: selectExtraServiceLabelValue
      vmstorage:
        name: storage
        fullnameOverride: vmstorage-node
        extraLabels:
          extraLabelName: storageExtraLabelValue
        service:
          labels:
            extraServiceLabelName: storageExtraServiceLabelValue
    asserts:
      - matchSnapshot: {}
