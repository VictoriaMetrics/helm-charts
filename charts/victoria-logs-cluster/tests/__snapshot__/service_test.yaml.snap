service should match snapshot:
  1: |
    apiVersion: v1
    kind: Service
    metadata:
      labels:
        app: vlinsert
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: victoria-logs-cluster
        app.kubernetes.io/version: 0.1.0-victorialogs
        helm.sh/chart: victoria-logs-cluster-0.1.1
      name: RELEASE-NAME-victoria-logs-cluster-vlinsert
      namespace: NAMESPACE
    spec:
      ports:
        - name: http
          port: 9481
          protocol: TCP
          targetPort: http
      selector:
        app: vlinsert
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/name: victoria-logs-cluster
      type: ClusterIP
  2: |
    apiVersion: v1
    kind: Service
    metadata:
      labels:
        app: vlselect
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: victoria-logs-cluster
        app.kubernetes.io/version: 0.1.0-victorialogs
        helm.sh/chart: victoria-logs-cluster-0.1.1
      name: RELEASE-NAME-victoria-logs-cluster-vlselect
      namespace: NAMESPACE
    spec:
      ports:
        - name: http
          port: 9471
          protocol: TCP
          targetPort: http
      selector:
        app: vlselect
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/name: victoria-logs-cluster
      type: ClusterIP
  3: |
    apiVersion: v1
    kind: Service
    metadata:
      labels:
        app: vlstorage
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: victoria-logs-cluster
        app.kubernetes.io/version: 0.1.0-victorialogs
        helm.sh/chart: victoria-logs-cluster-0.1.1
      name: RELEASE-NAME-victoria-logs-cluster-vlstorage
      namespace: NAMESPACE
    spec:
      clusterIP: None
      ports:
        - name: http
          port: 9491
          protocol: TCP
          targetPort: http
      selector:
        app: vlstorage
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/name: victoria-logs-cluster
      type: ClusterIP
  4: |
    apiVersion: v1
    kind: Service
    metadata:
      labels:
        app: vmauth
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: victoria-logs-cluster
        app.kubernetes.io/version: v1.116.0
        helm.sh/chart: victoria-logs-cluster-0.1.1
      name: RELEASE-NAME-victoria-logs-cluster-vmauth
      namespace: NAMESPACE
    spec:
      ports:
        - name: http
          port: 8427
          protocol: TCP
          targetPort: http
      selector:
        app: vmauth
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/name: victoria-logs-cluster
      type: ClusterIP
service should match snapshot with fullnameOverride and extraLabels:
  1: |
    apiVersion: v1
    kind: Service
    metadata:
      labels:
        app: vlinsert
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: victoria-logs-cluster
        app.kubernetes.io/version: 0.1.0-victorialogs
        helm.sh/chart: victoria-logs-cluster-0.1.1
      name: RELEASE-NAME-victoria-logs-cluster-vlinsert
      namespace: NAMESPACE
    spec:
      ports:
        - name: http
          port: 9481
          protocol: TCP
          targetPort: http
      selector:
        app: vlinsert
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/name: victoria-logs-cluster
      type: ClusterIP
  2: |
    apiVersion: v1
    kind: Service
    metadata:
      labels:
        app: vlselect
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: victoria-logs-cluster
        app.kubernetes.io/version: 0.1.0-victorialogs
        helm.sh/chart: victoria-logs-cluster-0.1.1
      name: RELEASE-NAME-victoria-logs-cluster-vlselect
      namespace: NAMESPACE
    spec:
      ports:
        - name: http
          port: 9471
          protocol: TCP
          targetPort: http
      selector:
        app: vlselect
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/name: victoria-logs-cluster
      type: ClusterIP
  3: |
    apiVersion: v1
    kind: Service
    metadata:
      labels:
        app: vlstorage
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: victoria-logs-cluster
        app.kubernetes.io/version: 0.1.0-victorialogs
        helm.sh/chart: victoria-logs-cluster-0.1.1
      name: RELEASE-NAME-victoria-logs-cluster-vlstorage
      namespace: NAMESPACE
    spec:
      clusterIP: None
      ports:
        - name: http
          port: 9491
          protocol: TCP
          targetPort: http
      selector:
        app: vlstorage
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/name: victoria-logs-cluster
      type: ClusterIP
  4: |
    apiVersion: v1
    kind: Service
    metadata:
      labels:
        app: auth
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: victoria-logs-cluster
        app.kubernetes.io/version: v1.116.0
        extraServiceLabelName: authExtraServiceLabelValue
        helm.sh/chart: victoria-logs-cluster-0.1.1
      name: vmauth-node
      namespace: NAMESPACE
    spec:
      ports:
        - name: http
          port: 8427
          protocol: TCP
          targetPort: http
      selector:
        app: auth
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/name: victoria-logs-cluster
      type: ClusterIP
